### DISCUSSION TITLE
CSV Plugin load with Birt RE API

### DISCUSSION THREAD

**Discussion by vartikashri** [eclipse-birt/birt#1804] [General]
Hi,

I am trying to load a customized CSV plugin to application however I am using Birt RE as API hence there is no plugin folder.

I already referred birt website documentation but it did not make me clear guidance about the way to load the plugins.

-------------------
I am using BIRT API 4.4.2 version 
Using org.eclipse.birt.report.engine.emitter.csv (custom plugin) which used to work within plugin directory of Birt->ReportEngine.
Now I am using Birt API (hence no more birt installation setup required) & plugin is **not** pluggable if I am giving it as dependency only.
plugin  xml : 
<plugin>
   <extension
         name="Engine CSV Emitters"
         point="org.eclipse.birt.report.engine.emitters">
      <emitter
            class="org.eclipse.birt.report.engine.emitter.csv.CSVReportEmitter"
            format="csv"
            id="org.eclipse.birt.report.engine.emitter.csv"
            mimeType="text/csv"/>
   </extension>

</plugin>

Do I need to convert it into java class?

I tried to load plugin with classpath too :
  dependencies {
      classpath("com.dcisc.birt.plugins:csv-emitter:1.2.0-DCIS-50823_birt_plugin_test-a11ce90-5-dev")
    }
still it's NOT getting load. 

Please help me out if someone tried same scenario before.

**Comment by hvbtup** [comment]
If you are using the (non-OSGI) BIRT-runtime, then maybe it suffices if you add your plugin JAR to the classpath (not tested). If your BIRT integration works somehow else, I can't help you (but maybe others?).

**Reply by vartikashri** [reply]
Yes it worked after adding jar into classpath.. Thanks for your reply!
